### DISCUSSION TITLE
Isolating face detection and question regarding descriptor extraction

### DISCUSSION THREAD

**Discussion by mayankagarwals** [vladmandic/human#107] [Q&A] [answered]
1. The current HSE facerec model computes age and gender along with descriptors. Do you think this still runs faster than the previous model which used to get just the descriptor

2. How does the descriptor work even without extracting the mesh?. I thought getting the landmarks was a pre-requisite. Or is another default landmarks model used in case we opt out from the 468 point mesh

**Comment by vladmandic** [comment, marked as answer]
> The current HSE facerec model computes age and gender along with descriptors. Do you think this still runs faster than the previous model which used to get just the descriptor

in my tests, yes, it's faster.

> How does the descriptor work even without extracting the mesh?. I thought getting the landmarks was a pre-requisite. Or is another default landmarks model used in case we opt out from the 468 point mesh

face descriptor is basically a feature vector created by a model before applying mathching params  
basically, any image classification or detection model creates features vectors internally as intermediate result before actually analyzing them further and returning result  
so question is just choosing which model internally creates feature vectors that i like as face descriptors  
- in case of `human`, i tried several and so far i like intermediate results of `hse-facerec` the most
- in case of `face-api`, its intermedia results of mesh model (with additional weights applied)  
  it's not that it actually needs mesh points

**Reply by mayankagarwals** [reply]
> in my tests, yes, it's faster

Great, thank you for clarifying!



> * in case of `human`, i tried several and so far i like intermediate results of `hse-facerec` the most

Got it. This clarifies everything for me. Thank you!
